### PR TITLE
PUB-3794 | Set even height for calendar rows in month view

### PR DESCRIPTION
**Pull request by MayaUribe**
### Description
Set even height for calendar rows in month view

### Context & Notes

[PUB-3794](https://buffer.atlassian.net/browse/PUB-3794): Calendar row heights vary depending on browser.

### Screenshots

**Before**

**After**
<div style="display:flex;">
  

  
</div>

### Best Practices Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [x] I have tested different user stories.
-   [x] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

### ⚠️ Pre-deploy Checklist

- [ ] I have manually loaded the URL of the staging deployment and confirmed it's working (see below**)
- [ ] I have confirmed that the Cypress E2E test is passing (_Note: we have a known issue where it doesn't run on some PRs. Hopefully this will be fixed soon._)
- [ ] I have confirmed that all other tests are passing.

** To visit the URL of the staging deployment, please vlick the link in the bot comment, or look under checks at the bottom of this PR and click "Details" next to `bufferbotbrains/cicd-buffer-publish`
image

### CODE REVIEW

**Review by anafilipadealmeida — COMMENTED** [review]
Hey @MayaUribe , I believe there's a permanent vertical scroll in Firefox 🤔

**Comment by MayaUribe** [comment]
> Hey @MayaUribe , I believe there's a permanent vertical scroll in Firefox 🤔

Yes good one @anafilipadealmeida I didn't notice it 🤔 it's been quite tricky to figure out for FF 🤔

**Comment by federicoweber** [comment]
Hey @MayaUribe and @anafilipadealmeida I think I got a good fix for the scrollbar issue in Firefox.

I've also tried to enforce the header height to 32px as per specs, which is working well in Chrome but Firefox is not fully respecting the height limit for table elements, so it's growing the header just a little bit (see attached screen shots, left is Firefox right is Chrome). I think this is as good as we can get with a table structure, if we want to get more precise we should consider using a div structure with a flexbox grid.


image

File: packages/calendar/components/DaysOfMonth/index.jsx [hunk 1]
```diff
@@ -19,7 +19,10 @@ import dayCells from '../Days';
 import getTimezoneFromBrowser from '../../utils/getTimezoneFromBrowser';
 
 const Row = styled.tr`
-  height: calc(100% / ${props => props.numberWeeks});
+  // We are adding 1 to the calculation to account for the header
+  // this theoretically will set the height to be smaller than needed but then
+  // the css will grow the elements proportionally to fill the space.
+  height: calc(100% / (${props => props.numberWeeks} + 1));
```
Comment (MayaUribe) [inline packages/calendar/components/DaysOfMonth/index.jsx:25]: 👀 Ah interesting @federicoweber!

**Comment by MayaUribe** [comment]
> Hey @MayaUribe and @anafilipadealmeida I think I got a good fix for the scrollbar issue in Firefox.
> 
> I've also tried to enforce the header height to 32px as per specs, which is working well in Chrome but Firefox is not fully respecting the height limit for table elements, so it's growing the header just a little bit (see attached screen shots, left is Firefox right is Chrome). I think this is as good as we can get with a table structure, if we want to get more precise we should consider using a div structure with a flexbox grid.
> 
> image

Good one @federicoweber the reason I didn't set a fixed height for the header is that it might grow in the future when we introduce the campaign's feature (the horizontal bars in the header), but I honestly think this could work for now since styling the height for tables in Firefox has proven to be complex 😅
@anafilipadealmeida do you have any thoughts?

**Review by anafilipadealmeida — APPROVED** [review]
Great thoughts here @federicoweber and @MayaUribe! I'm good with shipping this as it is if you are! 😄 
Yes, when it comes to the calendar structure I'm afraid we'll always have to make trade-offs. We started with a CSS grid structure but then switched to a table because we couldn't implement advanced scroll there. We need to account for accessibility, expandable cell rows and header (to see more campaigns and more posts), overflows.. So we might need to adjust this along the way and perhaps diverge slightly from specs at times 😅

**Comment by MayaUribe** [comment]
> Great thoughts here @federicoweber and @MayaUribe! I'm good with shipping this as it is if you are! 😄
> Yes, when it comes to the calendar structure I'm afraid we'll always have to make trade-offs. We started with a CSS grid structure but then switched to a table because we couldn't implement advanced scroll there. We need to account for accessibility, expandable cell rows and header (to see more campaigns and more posts), overflows.. So we might need to adjust this along the way and perhaps diverge slightly from specs at times 😅

Thank you for the context @anafilipadealmeida! That makes a lot of sense! I'm happy with this approach for now, and we'll adjust the header later when we need to include campaigns, at least it's close to the design 😅

**Pull request merged**